plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -291,7 +293,14 @@ private static String buildSplitPath(String tableLocation, AddFileEntry addActio
     {
         // paths are relative to the table location and are RFC 2396 URIs
         // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file
-        String path = URI.create(addAction.getPath()).getPath();
+        URI uri = URI.create(addAction.getPath());
+        String path = uri.getPath();
+
+        // org.apache.hadoop.fs.azurebfs.AzureBlobFileSystem encodes the path as URL
+        // https://issues.apache.org/jira/browse/HADOOP-18580
```

Comment:
> You mean that addAction.path content is URL-encoded in case of ABFS and not url encoded otherwise (HDFS, S3, ADLS, GS)?

Not exactly. addAction.path is **_URI_**-encoded. This is defined in Delta protocol https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file. _**URL**_-encoding happens in the subsequent logic when  trying to read the file on ABFS.